From 0ce63a9d140b82cdec09607d57fe578e3b8e6e4e Mon Sep 17 00:00:00 2001
From: Jason Lin <lin.jasonyy@inventec.com>
Date: Thu, 15 Dec 2022 16:18:28 +0800
Subject: [PATCH] Bug 1469 - [Starscream-ast][OpenBMC][WebUI] Sessions page
 does not show IP address correctly

Symptom/Reason:
	The structure of the ip address has changed.

Root Cause:
	The structure of the ip address has changed.

Solution/Change:
	Parse the ip address & btw the client id correctly.

Entry Test:
	https://${ip}/#/security-and-access/sessions
---
 src/store/modules/SecurityAndAccess/SessionsStore.js | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/store/modules/SecurityAndAccess/SessionsStore.js b/src/store/modules/SecurityAndAccess/SessionsStore.js
index 37ba9fc..a404767 100644
--- a/src/store/modules/SecurityAndAccess/SessionsStore.js
+++ b/src/store/modules/SecurityAndAccess/SessionsStore.js
@@ -25,14 +25,10 @@ const SessionsStore = {
         )
         .then((sessionUris) => {
           const allConnectionsData = sessionUris.map((sessionUri) => {
-            //For filtering IP address to IPv4
-            let filteredIPAddress = sessionUri.data?.ClientOriginIPAddress.slice(
-              7
-            );
             return {
-              clientID: sessionUri.data?.Oem?.OpenBMC.ClientID,
+              clientID: sessionUri.data?.Id,
               username: sessionUri.data?.UserName,
-              ipAddress: filteredIPAddress,
+              ipAddress: sessionUri.data?.ClientOriginIPAddress,
               uri: sessionUri.data['@odata.id'],
             };
           });
